nginx.tmpl

```diff
@@ -85,8 +85,8 @@ server {
 {{ end }}
 
 {{ range $host, $containers := groupByMulti $ "Env.VIRTUAL_HOST" "," }}
-
-upstream {{ $host }} {
+{{ $upstream_name := sha1 $host }}
+upstream {{ $upstream_name }} {
```

Comment:
Can you add `# {{ $host }}` just above the upstream declaration?  That will help with troubleshooting config issues.